cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -400,6 +420,28 @@ public void stop(ProgramId programId, @Nullable String runId) throws Exception {
     return runtimeInfo.getController().stop();
   }
 
+  /**
+   * Save runtime arguments for all future runs of this program. The runtime arguments are saved in the
```

Comment:
that's just the UI though. Other clients (e.g. CLI) still use this API and persist runtime arguments (although this goes to preferences store). Not sure we want to deprecate this API.